R/derive_vars_trt_start.R

```diff
@@ -0,0 +1,158 @@
+#'
+#' @param filter_ex Filter condition for the ex dataset
+#'
+#'   Only observations of the ex dataset which fulfill the specified condition
+#'   are considered for the treatment start date.
+#'
+#'   Default: NULL
+#'
+#'   Permitted Values: logical expression
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#' A list of quosures where the expressions are symbols as returned by
+#' `vars()` is expected.
+#'
+#' @param new_var name of treatment start date e.g. `TRTSDTM`
+#'
+#' @param dtc name of variable from which start date is derived e.g. `EXSTDTC`
+#'
+#' @param date_imputation The value to impute the day/month when a datepart is
+#'   missing.
+#'
+#'   If `NULL`: no date imputation is performed and partial dates are returned as
+#'   missing.
+#'
+#'   Otherwise, a character value is expected, either as a
+#'   - format with month and day specified as 'mm-dd': e.g. `"06-15"` for the 15th
+#'   of June,
+#'   - or as a keyword: `"FIRST"`, `"MID"`, `"LAST"` to impute to the first/mid/last
+#'   day/month.
+#'
+#'   Default is `NULL`.
+#'
+#' @param time_imputation The value to impute the time when a timepart is
+#'   missing.
+#'
+#'   A character value is expected, either as a
+#'   - format with hour, min and sec specified as 'hh:mm:ss': e.g. `"00:00:00"`
+#'   for the start of the day,
+#'   - or as a keyword: `"FIRST"`,`"LAST"` to impute to the start/end of a day.
+#'
+#'   Default is `"00:00:00"`.
+#'
+#' @param min_dates Minimum dates
+#'  A list of dates is expected. It is ensured that the imputed date is not
+#'  before any of the specified dates, e.g., that the imputed adverse event start
+#'  date is not before the first treatment date. Only dates which are in the
+#'  range of possible dates of the dtc value are considered. The possible dates
+#'  are defined by the missing parts of the dtc date (see example below). This
+#'  ensures that the non-missing parts of the dtc date are not changed.
+#'
+#' @param max_dates Maximum dates
+#'
+#' A list of dates is expected. It is ensured that the imputed date is not after
+#' any of the specified dates, e.g., that the imputed date is not after the data
+#' cut off date. Only dates which are in the range of possible dates are
+#' considered.
+#'
+#' @param ord_vars
+#'  Order of variables for which either the last or first observation
+#'  is picked as treatment start date
+#'  e.g. vars(EXSTDTC,EXSEQ)
+#'
+#' @param ord_filter
+#'  The mode by which the filter_extreme is chosen.
+#'  If "first", the first observation of each by group is included in the output dataset.
+#'  If "last", the last observation of each by group is included in the output dataset.
+#'
+#'  Permitted Values: "first", "last"
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` parameter) the first observation (with respect to the order
+#'   specified for the `order` parameter) is included in the output dataset.
```

Comment:
I would suggest something like : For each group (with respect to the variables specified for the `by_vars` parameter), the first or last (with respect to the order specified for the `order` parameter)  `dtc`  is selected and used as input for 'new_var'. We should also mention about the imputation variables.